### PR TITLE
Update `maxAge`-behaviour for `/verify` & `/retrieve`

### LINKED ISSUE #216: Allignment on `maxAge` - behaviour for location-verification & location-retrieval
**Problem description**
In `location-retrieval` it is described that the application throws a HTTP-422 if the requested `maxAge` does not fit with the freshness of the location provided by the network.

#### 1. Issue
In location-retrieval there is still a HTTP-400 - Bad Request when the maxAge treshold cannot be satisfied:
This example needs to be removed them: `MaxAgeIssue` as this conflicts with `HTTP-422`.

```yaml
    RetrieveLocationBadRequest400:
      description: |-
        Problem with the client request. In addition to regular scenario of `INVALID_ARGUMENT`, another scenarios may exist:
          - maxAge threshold cannot be satisfied
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/ErrorInfo'
          examples:
             #...
            MaxAgeIssue:
              value:
                status: 400
                code: LOCATION_RETRIEVAL.MAXAGE_INVALID_ARGUMENT
                message: "maxAge threshold cannot be satisfied"
```

#### 2. Issue
The behaviour of freshness is just described for location-retrieval.
What about location-verification? There also the maxAge can be set and checks for the threshold.
But here just HTTP-400 is provided and not HTTP-422.

**Expected behavior**
1. Cleanup the `MaxAgeIssue` example in location-retrieval
2. Add HTTP-422 - behavior for location-verification.

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:
* correction

#### What this PR does / why we need it:
Updating the `maxAge`-behaviour for location-verification and location-retrieval

#### Which issue(s) this PR fixes:

Fixes #216

### CODE REVIEW

File: code/API_definitions/location-verification.yaml [hunk 1]
```diff
@@ -162,6 +162,8 @@ paths:
           $ref: "#/components/responses/Generic403"
         "404":
           $ref: "#/components/responses/Generic404"
+        "422":
+          $ref: "#/components/responses/RetrieveLocationUnprocessableEntity422"
```
Comment (mhfoo) [inline code/API_definitions/location-verification.yaml:166]: The current YAML file is location-verification.yaml, should this be VerifyLocationUnprocessableEntity422?

Comment (maxl2287) [reply code/API_definitions/location-verification.yaml:166]: Good catch!
Thanks @mhfoo !

**Comment by maxl2287** [comment]
@camaraproject/device-location_maintainers this PR is ready to be reviewed 😸

**Review by bigludo7 — APPROVED** [review]
LGTM/

**Review by bigludo7 — APPROVED** [review]

File: code/API_definitions/location-verification.yaml [hunk 2]
```diff
@@ -456,6 +458,19 @@ components:
             status: 404
             code: NOT_FOUND
             message: "The specified resource is not found"
+    VerifyLocationUnprocessableEntity422:
```
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:461]: This is in contradiction with L38:
maxAge=0 means a fresh calculation is requested by the client. If the system does not support it, or fresh location cannot be checked at that time for any reason, the API response will be "UNKNOWN" and the lastLocationTime attribute may indicate the last available time for the device location.

The is also an ongoing discussion for this in the related issue

File: code/API_definitions/location-verification.yaml [hunk 3]
```diff
@@ -456,6 +458,19 @@ components:
...
+      headers:
+        x-correlator:
+          $ref: '#/components/headers/x-correlator'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorInfo'
+          example:
+            status: 422
+            code: LOCATION_RETRIEVAL.UNABLE_TO_FULFILL_MAX_AGE
```
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:472]:
```suggestion
            code: LOCATION_VERIFICATION.UNABLE_TO_FULFILL_MAX_AGE
```

File: code/API_definitions/location-verification.yaml (hunk 2 above)
Comment (jlurien) [reply code/API_definitions/location-verification.yaml:461]: Decision during last meeting was to apply 422 to verification as well, but we have to update description (L38) accordingly

**Review by jlurien — CHANGES_REQUESTED** [review]

**Comment by jlurien** [comment]
@maxl2287 please give priority to this PR as it is blocking the process for these 2 APIs. Thanks

**Comment by maxl2287** [comment]
Update is done.
Please check @jlurien and @bigludo7

**Review by bigludo7 — COMMENTED** [review]
Hello @maxl2287 
Sorry if I miss something but I do not understand why you removed all 422 codes
For me we have to keep these: 


```
            GENERIC_422_UNPROCESSABLE_ENTITY:
              summary: Unprocessable entity
              description: The request was well-formed but was unable to be processed due to semantic errors or not applicable values. This is the generic error code for 422 responses.
              value:
                status: 422
                code: UNPROCESSABLE_ENTITY
                message: "Value not acceptable: ..."
            GENERIC_422_DEVICE_NOT_APPLICABLE:
              summary: Service not applicable to the device
              description: The provided device is not compatible with the requested operation, according to the service provider rules.
              value:
                status: 422
                code: DEVICE_NOT_APPLICABLE
                message: "The device is not applicable for the requested operation"
            GENERIC_422_DEVICE_IDENTIFIERS_MISMATCH:
              summary: Device identifiers mismatch
              description: Several device identifiers are provided but do not match the same device
              value:
                status: 422
                code: DEVICE_IDENTIFIERS_MISMATCH
                message: "The provided device identifiers do not match the same device"
            GENERIC_422_UNSUPPORTED_DEVICE_IDENTIFIERS:
              summary: None of the provided device identifiers is supported by the implementation
              description: Message may list the supported device identifiers
              value:
                status: 422
                code: UNSUPPORTED_DEVICE_IDENTIFIERS
                message: "Supported device supported are: ..."
            GENERIC_422_UNIDENTIFIABLE_DEVICE:
              summary: No identifier provided
              description: No device identifier provided for the device to be located
              value:
                status: 422
                code: UNIDENTIFIABLE_DEVICE
                message: "A device must be provided"
```

**Comment by jlurien** [comment]
We have to add another example to the examples array, e.g.:
`
           LOCATION_VERIFICATION_422_UNABLE_TO_FULFILL_MAX_AGE:
              summary: Unable to fulfill maxAge
              description: the system is not able to provide the fresh location required by the client
              value:          
                 status: 422
                 code: LOCATION_VERIFICATION.UNABLE_TO_FULFILL_MAX_AGE
                 message: "Unable to provide expected freshness for location"

File: code/API_definitions/location-verification.yaml [hunk 4]
```diff
@@ -536,10 +536,49 @@ components:
...
           schema:
             $ref: "#/components/schemas/ErrorInfo"
-          example:
-            status: 422
-            code: LOCATION_VERIFICATION.UNABLE_TO_FULFILL_MAX_AGE
-            message: "Unable to provide expected freshness for location"
+          examples:
+            LOCATION_VERIFICATION_422_UNABLE_TO_FULFILL_MAX_AGE:
+              summary: Unable to fulfill maxAge
+              description: the system is not able to provide the fresh location required by the client
```
Comment (jlurien) [inline code/API_definitions/location-verification.yaml:542]:
```suggestion
              description: The system is not able to provide the fresh location required by the client
```

**Review by jlurien — DISMISSED** [review]
LGTM

**Review by jlurien — APPROVED** [review]

**Comment by jlurien** [comment]
@bigludo7 take a final look and we can merge it. Thanks

**Review by bigludo7 — APPROVED** [review]
The 422 examples are back :)

**Pull request merged**